onformsubmit-http/utils/firecloud.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Handles Firecloud calls
+ * @module Firecloud
+ */
+
+/** Sends HTTP requests */
+class Firecloud {
+  /**
+   * @param {Object} axios [the configured http client]
+   */
+  constructor(axios) {
+    this._axios = axios;
+
+    this._groupEmail = undefined;
+  }
+
+  /**
+   * Creates a new Terra group and adds users
+   * @param {string} groupName [the name of the group to create]
+   * @param {Array.<string>} userEmails [the description of the group to create]
+   */
+  createGroup = async (groupName) => {
+    try {
+      await this._axios.post(`/api/groups/${groupName}`).then((res) => {
+        this._groupEmail = res.data.groupEmail;
+      });
+    } catch (e) {
+      throw new Error(e);
+    }
+
+    // if group wasn't created, throw an error
+    if (this._groupEmail) {
+      return this._groupEmail;
+    } else {
+      throw new Error("Group was not created");
+    }
+  };
+
+  /**
+   * Add a user to a group
+   * - role: ['member', 'admin']
+   * @param {string} groupName [the name of the group]
+   * @param {string} userEmail [the email of the user to add]
+   * @param {string} [role='member'] [the role to assign to the user]
+   */
+  addUserToGroup = async (groupName, userEmail, role = "member") => {
+    try {
+      await this._axios.put(`/api/groups/${groupName}/${role}/${userEmail}`);
+    } catch (e) {
+      throw new Error(e);
+    }
+  };
+
+  /**
+   * Removes a user from a group
+   * @param {string} groupName [the name of the group]
+   * @param {string} userEmail [the email of the user to remove]
+   * @param {string} [role='member'] [the role of the user to be removed]
+   */
+  removeUserFromGroup = async (groupName, userEmail, role = "member") => {
+    try {
+      await this._axios.delete(`/api/groups/${groupName}/${role}/${userEmail}`);
+    } catch (e) {
+      throw new Error(e);
+    }
+  };
+
+  /**
+   * Create workspace and shares with users
+   * @param {string} workspaceName [the name of the workspace to create]
+   * @param {string} billingProject [the billing project to use]
+   * @param {string} authDomain [the auth domain to use]
+   */
+  createWorkspace = async (workspaceName, billingProject, authDomain) => {
+    const workspaceRequest = {
+      name: workspaceName,
+      namespace: billingProject,
+      authorizationDomain: [
+        {
+          membersGroupName: authDomain,
+        },
+      ],
+      attributes: {},
+      noWorkspaceOwner: false,
+    };
+
+    try {
+      await this._axios.post("/api/workspaces", workspaceRequest);
+    } catch (e) {
+      throw new Error(e);
```

Comment:
Maybe consider logging to console when these errors happen? I think the function may already if there is an error thrown, but it would be nice to have messages that give a better indicator of where/what is going on when looking at them in stackdriver.

Reply:
when the firecloud object throws these errors, they get caught by the method used and logged in the stackdriver